src/select/examples.js

```diff
@@ -11,315 +11,164 @@ LICENSE file in the root directory of this source tree.
-const emptyOption = {id: '', label: ''};
-
-class ParentSearch extends React.Component<ExamplePropsT, ExampleStateT> {
-  static defaultProps: {} = {};
-  constructor(props: ExamplePropsT) {
-    super(props);
-    this.state = {
-      options: props.filterable ? generateOptions('text to search') : [],
-      selectedOptions: [
-        {
-          id: '123',
-          label: 'label for 123',
-        },
-      ],
-    };
+const onChange = (e, {type, option = emptyOption}) => {
+  if (type === STATE_CHANGE_TYPE.select) {
+    // eslint-disable-next-line no-console
+    console.log('Selected option:' + option.id);
   }
-  render() {
-    return (
-      <React.Fragment>
-        <StatefulSelect
-          rows={8}
-          options={query => {
-            return new Promise(resolve => {
-              if (!query) {
-                resolve(this.state.options);
-              } else {
-                let text = query;
-                let options = [];
-                if (text.length > 5) {
-                  setTimeout(() => {
-                    options = generateOptions(text);
-                    this.setState({options: options}, () =>
-                      resolve(this.state.options),
-                    );
-                  }, 1500);
-                } else {
-                  resolve(this.state.options);
-                }
-              }
-            });
-          }}
-          multiple={true}
-          filterable={this.props.filterable}
-          type={TYPE.search}
-          initialState={{
-            selectedOptions: this.state.selectedOptions,
-          }}
-          label="Search for tags"
-          placeholder="Start searching"
-          onTextInputChange={(e: SyntheticEvent<HTMLInputElement>) => {
-            //$FlowFixMe
-            console.log('Text has changed to:' + e.target.value);
-          }}
-          onChange={(e, {type, option = emptyOption, selectedOptions}) => {
-            switch (type) {
-              case STATE_CHANGE_TYPE.select:
-                // eslint-disable-next-line no-console
-                console.log('Selected id:' + option.id);
-                break;
-              case STATE_CHANGE_TYPE.unselect:
-                // eslint-disable-next-line no-console
-                console.log('Unselected id:' + option.id);
-                break;
-            }
-          }}
-        />
-      </React.Fragment>
-    );
-  }
-}
```

Comment:
Thee `Parent*` were helpful when we were auto-generating options but are less helpful after the refactor. The new examples just define their Selects directly (reusing change handlers where applicable), and the source code for each story still ends up being quite short (~10 or so lines).

These ParenSelect's also led to sort of misleading examples in the Storybook "Show Info" plugin
![image](https://user-images.githubusercontent.com/875591/46427132-69da4700-c6f5-11e8-97d0-f93bca517b7c.png)